src/http.rs

```diff
@@ -38,12 +38,35 @@ pub fn client(feature: Option<&str>) -> Client {
 
 pub fn auth_client(feature: Option<&str>, user: &GlobalUser) -> Client {
     let mut headers = headers(feature);
-    headers.insert("X-Auth-Key", HeaderValue::from_str(&user.api_key).unwrap());
-    headers.insert("X-Auth-Email", HeaderValue::from_str(&user.email).unwrap());
+    let auth_headers = add_auth_headers(&mut headers, user);
 
     builder()
-        .default_headers(headers)
+        .default_headers(auth_headers.to_owned())
         .redirect(RedirectPolicy::none())
         .build()
         .expect("could not create authenticated http client")
 }
+
+fn add_auth_headers<'a>(headers: &'a mut HeaderMap, user: &GlobalUser) -> &'a HeaderMap {
```

Comment:
i understand that `<'a>` is a lifetime - could you explain a bit why they're needed here and what they're doing?

Reply:
ah! we want to edit the preexisting headers struct instead of making a clone of it (in general this is just better/cleaner memory-wise). This did remind me though, we don't need to even return headers here; I've removed this return.